src/_lib/repl/index.ts

```diff
@@ -38,46 +28,41 @@ const promisableEval: REPLEval = (cmd, context, filename, callback) => {
   return callback(null, result);
 };
 
-const makeREPL = ({
-  container,
-  config: {
-    appName,
-    cli,
-    environment,
-    repl: { port },
-  },
-  logger,
-}: ReplParameters): REPL => {
+const makeREPL = ({ context, prompt, cli, remote, logger }: REPLProps): REPLInstance => {
   let server: Server;
 
   const create = (config: Partial<ReplOptions> = { input: process.stdin, output: process.stdout }): REPLServer => {
     const repl = REPL.start({
       eval: promisableEval,
-      prompt: `${appName}$ `,
+      prompt: `${prompt}$ `,
       ignoreUndefined: true,
       ...config,
     });
 
-    Object.assign(repl.context, { registry: container.cradle, container });
+    Object.assign(repl.context, context);
 
     return repl;
   };
 
+  let destroySocket: (...args: any) => void = () => null;
```

Comment:
Are we able to use the socket type itself here?

```suggestion
  let destroySocket: Socket['destroy'] = () => null;
```